### PR TITLE
perf(sources): no-fsync census spill on data disk + decoded-session RAM layer

### PR DESCRIPTION
**Pull request by Sinity**
## Summary
Cuts the two dominant non-write stages of the serial backfill engine pass (measured live via #3206 telemetry: `census=135.0s spill_load=105.2s` vs ~50-100s of index writes per 2,000-raw page).

## Problem
The census spill is a disposable parse cache, but it paid durable-storage costs: a synchronous journal commit per censused raw, placed under `archive_root` — which on symlinked deployments is the wear-limited root SSD, not the NVMe the tier symlinks point at. Replay then re-read what census had just parsed through sqlite + `pickle.loads`.

## Solution
`_ParsedSessionSpill` only: `journal_mode=OFF`/`synchronous=OFF`/`temp_store=MEMORY` (loss = reparse durable source bytes, by design); spill file placed beside the resolved `index.db`; a 256MiB payload-bounded decoded-session dict layer in front of the pickle path (cohorts are consumed almost immediately after census).

## Verification
`devtools test tests/unit/sources/test_revision_backfill.py tests/unit/storage/test_revision_replay.py` → 59 passed; mypy/ruff clean. Live receipt: next instrumented page's stage-timing line will be quoted here for before/after.

### CODE REVIEW

**Comment by Sinity** [comment]
Superseded by #3208 (same commit included in its stack).
